src/fibsem_tools/io/fibsem_h5.py

```diff
@@ -0,0 +1,314 @@
+"""
+Utilities for loading and saving FIBSEM Data as HDF5 fliles
+
+High-level Interface
+* create_default_h5_files
+* create_zstd_h5_files
+* create_gzip_h5_files
+"""
+
+import h5py
+import numpy as np
+from typing import Iterable, Union, Sequence, Tuple
+import re
+import os.path
+import glob
+import hdf5plugin
+from .fibsem import (
+    FibsemDataset,
+    FIBSEMData,
+    FIBSEMHeader,
+    OFFSET,
+    MAGIC_NUMBER,
+    read_fibsem,
+)
+
+chunk_type = Union[Tuple[int, int], Tuple[int, int, int]]
+
+
+def create_fibsem_h5_dataset(
+    parent: Union[h5py.File, h5py.Group],
+    name: str,
+    data: FIBSEMData,
+    chunks: chunk_type = (128, 128),
+    **kwargs
+):
+    """
+    Create a HDF5 dataset from an existing FIBSEMData object.
+
+    :param parent: `h5py.File` or `h5py.Group` to place the dataset in
+    :param name: Name of the dataset
+    :param data: `FIBSEMData` instance
+    :param chunks: tuple describing the chunk size (Y, X)
+    :type chunks: tuple, optional
+    :param **kwargs: Keyword arguments are forwarded to `parent.create_dataset`
+    """
+
+    # If chunks has length 2, add a 1 for the ChanNum dimension
+    if len(chunks) == 2:
+        if data.shape[0] == data.header.ChanNum:
+            chunks = (1, *chunks)
+        elif data.shape[2] == data.header.ChanNum:
+            chunks = (*chunks, 1)
+
+    ds = parent.create_dataset(name, data.shape, data.dtype, chunks=chunks, **kwargs)
+    for key, value in data.header.__dict__.items():
+        ds.attrs[key] = value
+    ds[:] = data
+    return ds
+
+
+def create_fibsem_h5_dataset_external(
+    parent: Union[h5py.File, h5py.Group],
+    name: str,
+    data: FIBSEMData,
+    datfile: str,
+    **kwargs
+):
+    """
+    Create a HDF5 external dataset where the data is still stored in the dat file.
+    Note that the axes is rolled in the dat file versus FIBSEMData object
+
+    :param parent: `h5py.File` or `h5py.Group` to place the dataset in
+    :param name: Name of the dataset
+    :param data: `FIBSEMData` instance
+    :param **kwargs: Keyword arguments are forwarded to `parent.create_dataset`
+    """
+
+    # FIBSEMData is usually (ChanNum, YResolution, XResolution)
+    shape = (
+        data.header.YResolution,
+        data.header.XResolution,
+        data.header.ChanNum,
+    )
+    parent.create_dataset(
+        name,
+        data.shape,
+        ">i2",
```

Comment:
we can infer this from `data.type`

Reply:
Fixed in https://github.com/janelia-cosem/fibsem-tools/pull/18/commits/1465176d8446c8af22cf57393d6025e77cdd7cf7